smoke-test/src/test/java/org/opennms/smoketest/BSMAdminIT.java

```diff
@@ -1,88 +1,197 @@
 /*******************************************************************************
  * This file is part of OpenNMS(R).
- *
+ * <p>
  * Copyright (C) 2011-2015 The OpenNMS Group, Inc.
  * OpenNMS(R) is Copyright (C) 1999-2015 The OpenNMS Group, Inc.
- *
+ * <p>
  * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
- *
+ * <p>
  * OpenNMS(R) is free software: you can redistribute it and/or modify
  * it under the terms of the GNU Affero General Public License as published
  * by the Free Software Foundation, either version 3 of the License,
  * or (at your option) any later version.
- *
+ * <p>
  * OpenNMS(R) is distributed in the hope that it will be useful,
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU Affero General Public License for more details.
- *
+ * <p>
  * You should have received a copy of the GNU Affero General Public License
  * along with OpenNMS(R).  If not, see:
- *      http://www.gnu.org/licenses/
- *
+ * http://www.gnu.org/licenses/
+ * <p>
  * For more information contact:
- *     OpenNMS(R) Licensing <license@opennms.org>
- *     http://www.opennms.org/
- *     http://www.opennms.com/
+ * OpenNMS(R) Licensing <license@opennms.org>
+ * http://www.opennms.org/
+ * http://www.opennms.com/
  *******************************************************************************/
 
 package org.opennms.smoketest;
 
+import org.junit.After;
+import org.junit.Before;
 import org.junit.FixMethodOrder;
 import org.junit.Test;
 import org.junit.runners.MethodSorters;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
 
 @FixMethodOrder(MethodSorters.NAME_ASCENDING)
 public class BSMAdminIT extends OpenNMSSeleniumTestCase {
 
     private final String RENAMED_SERVICE_NAME = "renamed_service";
-    private final String BSM_LIST_URL = BASE_URL + "opennms/admin/bsm/index.jsp";
+    private final String BSM_ADMIN_URL = BASE_URL + "opennms/admin/bsm/adminpage.jsp";
     private final String BASIC_SERVICE_NAME = "BasicService";
-    private final String BASIC_SERVICE_NAME_TO_CHANGE = "BasicServiceToChange";
+
+    private final String ADD_BUTTON_XPATH = "/html/body/div[3]/div[5]/div/div/div[5]/div/div/div/div[3]/div/div[2]/div/div[2]/div[1]/span/span";
```

Comment:
I agree with Jesse. It is always better to minimize as much as you can either the XPath or the CSS Selector used to find an HTML element. Also, I would use the easier from both selection technologies.

Here is a hint. With FireBug, on the console, you can write something like $x("blahblah") where blah blah is an XPath, to verify your selection.